cpp/include/cuspatial/legacy/soa_readers.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+typedef struct gdf_column_ gdf_column; // forward declaration
+
+namespace cuspatial {
+
+/**
+ * @brief read uint32_t data from file as column
+ *
+ * @param[in] filename: file to read
+ *
+ * @return gdf_column storing the uint32_t data
+ **/
+gdf_column read_uint32_soa(const char *filename);
```

Comment:
While we're at it can we just get rid of the whole "soa" terminology? I don't think it's a standard geospatial thing. 

What file format is this reading? Is it a standard format? Can we replace it with the existing cuIO readers?

Reply:
Maybe changes at this level should be left after the initial port...

Reply:
They're just packed binary files with little structs to define the format. The original source are the `.locust` files used in the demos, and these i/o calls are still the basic data source for those demos of `cuspatial` functionality, absent any other i/o.

Maybe I should skip the soa i/o and do the shapefile i/o only, and move on to other core functionality? I'm not sure if another reader can be used for the original `.locust` files at all. @zhangjianting, could we read the original demo data with fiona if we wanted to?

Reply:
> They're just packed binary files with little structs to define the format. The original source are the `.locust` files used in the demos, and these i/o calls are still the basic data source for those demos of `cuspatial` functionality, absent any other i/o.
> 
> Maybe I should skip the soa i/o and do the shapefile i/o only, and move on to other core functionality? I'm not sure if another reader can be used for the original `.locust` files at all. @zhangjianting, could we read the original demo data with fiona if we wanted to?

The original file is a JSON file with id/lat/lon/timestamp fields. I believe cudf can handle the file and extract the fields.

Reply:
cuDF can read json_lines but not full json.

Reply:
Can you provide the url of the original JSON file? That'd be great!

Reply:
@zhangjianting See above.

Reply:
> Can you provide the url of the original JSON file? That'd be great!
@thomcom @harrism Sorry that I missed this. Please see https://github.com/rapidsai/cuspatial/tree/branch-0.14/data for URLs related to real data. 
For the particular Json file, it is at https://drive.google.com/file/d/1GKTB5SV2RK7lEOIWz8tWab5MGWDtMWWW/view?usp=sharing

Reply:
Awesome, thanks! Resolving this, see #181 for the implementations.